crates/runtime/src/dataconnector/parameters.rs

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::sync::Arc;
+
+use app::App;
+use datafusion_table_providers::UnsupportedTypeAction;
+use tokio::sync::RwLock;
+
+use crate::{
+    Runtime, catalogconnector::CATALOG_CONNECTOR_FACTORY_REGISTRY, get_params_with_secrets,
+    parameters::Parameters, secrets::Secrets,
+};
+
+use super::{
+    ConnectorComponent, DATA_CONNECTOR_FACTORY_REGISTRY, DataConnectorError, ODBC_DATACONNECTOR,
+};
+
+pub(crate) mod aws;
+
+#[derive(Clone)]
+pub struct ConnectorParams {
+    pub(crate) parameters: Parameters,
+    pub(crate) unsupported_type_action: Option<UnsupportedTypeAction>,
+    pub(crate) component: ConnectorComponent,
+    pub(crate) app: Option<Arc<App>>,
+    #[allow(dead_code)] // will be utilized in https://github.com/spiceai/spiceai/pull/5542
```

Comment:
Can we either remove this dead_code directive or remove the runtime here if its still not used?